packages/framework/presence/src/internalUtils.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Returns union of types of values in a record.
+ */
+export type RecordEntryTypes<T> = T[keyof T];
+
+type MapNumberIndicesToStrings<T> = {
+	[K in keyof T as K extends number ? `${K}` : K]: T[K];
+};
+
+type KeyValuePairs<T> = {
+	[K in keyof MapNumberIndicesToStrings<Required<T>>]: [K, Required<T>[K]];
+}[keyof MapNumberIndicesToStrings<Required<T>>][];
+
+type RequiredAndNotUndefined<T> = {
+	[K in keyof T]-?: Exclude<T[K], undefined>;
+};
+
+/**
+ * Object.entries retyped to preserve known keys and their types.
+ *
+ * @internal
+ */
+export const objectEntries = Object.entries as <T>(o: T) => KeyValuePairs<T>;
```

Comment:
should these helpers be migrated to core-utils?

Reply:
Maybe. I don't like to pre-emptively place utilities in a shared space without more than one use. Certainly, consider if these are generally useful as you look at fixing things. If they are, then we refactor.